File: jgrpc/src/main/java/jgrpc/lib/sym/Location.java
package jgrpc.lib.sym;

import org.antlr.v4.runtime.tree.TerminalNode;

public class Location {
    private int lineno;
    private int columnno;

    public Location(TerminalNode tn) {
        lineno = tn.getSymbol().getLine();
        columnno = tn.getSymbol().getCharPositionInLine();
    }

    @Override
    public String toString() {
        return String.format("%d:%d", lineno, columnno);
    }
}

File: grpc/src/main/kotlin/grpc/lib/symbol/Location.kt
package grpc.lib.symbol

import org.antlr.v4.runtime.Token
import org.antlr.v4.runtime.tree.TerminalNode

class Location(val line: Int, val column: Int) {
    constructor(token: Token?) : this(token!!.line, token.charPositionInLine)

    constructor(tn: TerminalNode?) : this(tn!!.symbol)

    override fun toString(): String = "$line:$column"
}